src/services/contracts/deployments.ts

```diff
@@ -0,0 +1,87 @@
+import semverSatisfies from 'semver/functions/satisfies'
+import {
+  getSafeSingletonDeployment,
+  getSafeL2SingletonDeployment,
+  getMultiSendCallOnlyDeployment,
+  getFallbackHandlerDeployment,
+  getProxyFactoryDeployment,
+  getSignMessageLibDeployment,
+} from '@safe-global/safe-deployments'
+import type { SingletonDeployment, DeploymentFilter } from '@safe-global/safe-deployments'
+import type { ChainInfo, SafeInfo } from '@safe-global/safe-gateway-typescript-sdk'
+
+import { LATEST_SAFE_VERSION } from '@/config/constants'
+
+export const _tryDeploymentVersions = (
+  getDeployment: (filter?: DeploymentFilter) => SingletonDeployment | undefined,
+  network: string,
+  version: SafeInfo['version'],
+): SingletonDeployment | undefined => {
+  // Get deployments using LATEST_SAFE_VERSION for unsupported frontends/versions
+
+  return (
+    getDeployment({
+      version: version || LATEST_SAFE_VERSION,
+      network,
+    }) ||
+    getDeployment({
+      version: version || LATEST_SAFE_VERSION,
+    }) ||
+    getDeployment({
+      version: LATEST_SAFE_VERSION,
+      network,
+    }) ||
+    getDeployment({
+      version: LATEST_SAFE_VERSION,
+    })
+  )
```

Comment:
It's a little hard to to follow with ORs on two levels. I would try to avoid `version || LATEST_SAFE_VERSION` and make it more explicit. E.g.

```js
if (version !== null) {
  return  getDeployment({
       version,
       network,
     }) ||
     getDeployment({
       version
     })
} else {
  return  getDeployment({
       version: LATEST_SAFE_VERSION,
       network,
     }) ||
     getDeployment({
       version: LATEST_SAFE_VERSION
    })
  )
```

Reply:
Do I get it right that the Core SDK will fail to init then? It won't crash the app tho, will it?

Reply:
> Do I get it right that the Core SDK will fail to init then? It won't crash the app tho, will it?

Yes. It's [caught during initialisation](https://github.com/safe-global/safe-wallet-web/blob/dev/src/hooks/coreSDK/useInitSafeCoreSDK.ts#L30-L50).